clients/drcachesim/scheduler/scheduler.cpp

```diff
@@ -1833,6 +1841,29 @@ scheduler_tmpl_t<RecordType, ReaderType>::next_record(output_ordinal_t output,
                     // over timestamp ordering.
                     need_new_input = true;
                 }
+            } else if (options_.quantum_unit == QUANTUM_TIME) {
+                // The above logic is for non-instrs, except the blocking syscall
+                // next instr which is already switching: so an else{} works here.
```

Comment:
I don't understand this comment.  Isn't the above logic mostly for instructions?

Reply:
Updated to say "The above if-else cases are all either for non-instrs or QUANTUM_INSTRUCTIONS, except ".  I spent a little time figuring out how to structure this addition to the sequence of cases here so I wanted a comment to make clear the logic for not having a separate top-level if() (still have to deal w/ blocking syscalls).